Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -487,10 +582,10 @@ private extension PurchasesOrchestrator {
         }
     }
 
-    // Although both SKProduct.productIdentifier and SKPayment.productIdentifier
+    // Although both SK1Product.productIdentifier and SKPayment.productIdentifier
     // are supposed to be non-null, we've seen instances where this is not true.
     // so we cast into optionals in order to check nullability, and try to fall back if possible.
-    func extractProductIdentifier(fromProduct product: SKProduct, orPayment payment: SKPayment) -> String? {
+    func extractProductIdentifier(fromProduct product: SK1Product, orPayment payment: SKPayment) -> String? {
```

Comment:
could we make this a function on the products themselves? maybe we still keep this method to choose one or the other, but each if/let and it's associated apple warning are in `SK1Product` and `SKPayment`

Reply:
that's a great idea. We didn't use to have a great place for it because we didn't have a wrapper for `SKProduct`, but `SK1ProductDetails` feels like a great place for this

Reply:
https://app.shortcut.com/revenuecat/story/11121/move-extractproductidentifier-to-sk1productdetails